sdgym/benchmark.py

```diff
@@ -149,11 +150,71 @@ def _score_with_timeout(timeout, synthesizer, metadata, metrics, iteration):
         return output
 
 
+def _write_to_cache_dir(cache_dir, name, dataset_name, iteration, run_id, scores,
```

Comment:
I'm thinking about changing this approach and, instead of writing a method that handles all the different situations, just write a `write_file(contents, path, aws_key, aws_secret)` function that knows whether to write in S3 or in a local file depending on the path prefix.
Additionally, a `write_csv` could be added that just dumps the DataFrame as a CSV string and then calls `write_file`.

With that, we can keep the code almost as it was before, doing:

```python3
base_path = str(cache_dir / f'{name}_{dataset_name}_{iteration}_{run_id}')
if scores is not None:
    write_csv(scores, base_path + '_scores.csv')
if 'synthetic_data' in output:
    synthetic_data = compress_pickle.dumps(output['synthetic_data'])
    write_file(synthetic_data, base_path + '.data.gz')
if 'exception' in output:
    write_file(output['exception'],base_path + '_error.txt')
```